external/mmat.m

```diff
@@ -51,9 +52,46 @@
 nDB = ndims(B);
 nD = max(nDA,nDB);
 
+if nD == 2 && isdefaultdim
+    C = A * B;
+    return
+end
+
 nA = [size(A),ones(1,nD-nDA)]; nA = nA(dim); 
 nB = [size(B),ones(1,nD-nDB)]; nB = nB(dim);
 
+neededMem = (numel(A)*nB(2) + numel(B)*nA(1)) * 16;
```

Comment:
Is this always true? I think the memory of required for an array is 
`numel(array)*bytes_per_element`
It seems that by default MATLAB uses double precision types uint64 & int64 with 8 bytes per element, but it is possible to have uint32 & int32 formats have 4 bytes per element - should we check?
Possibly a silly question, but where is the factor of `16` coming from? It looks like `sw_freemem` returns the memory of the maximum possible array allowed in bytes should it be 8 or 4?